004-governance.Rmd

```diff
@@ -0,0 +1,153 @@
+As well as the responsibilities of an author, they also:
+
+-   Set and clearly communicates the strategic objectives of the project.
+-   Oversee CRAN releases.
+-   On-board new authors.
+-   Mediate conflicts amongst authors.
+-   Enforce the code of conduct.
+-   Recruit their replacement if they want to retire from the project.
+
+A maintainer has two special powers:
+
+-   They have **admin** access on Github, allowing them to add new authors when needed.
+
+-   They are listed in `Authors@R`, along with their email address and the "cre" (creator) role (a maintainer often isn't the creator, but CRAN mandates the use of this role).
+
+Maintainer turnover is slow and we have not yet developed a process for it, but we'd generally expect a maintainer to be a long-standing author.
+
+The maintainers of most of RStudio's open source repositories are RStudio employees.
+This is not a requirement but a recognition of the tension between making development open to all and ensuring that users can trust that a package will be maintained in the long term (i.e. 10+ years), which typically requires the maintainer be explicitly remunerated for their work.
+Where the maintainer is not an employee of RStudio, we ask for the "right of first refusal" — if the maintainer wants to stop maintaining the package (for whatever reason) they first offer it back to RStudio.
+In the future, we hope to find other ways of financial supporting maintainers apart from full-time employment.
+
+### Common tasks
+
+#### Author invitation
+
+To on-board a new author, the maintainer looks for rough consensus amongst authors (which would include no objections within 7 days).
+The maintainer then sends the following email:
+
+> Hi {name},
+>
+> In recognition of your significant contributions to {package}, would you be interested in becoming a co-author of the package?
+> This means that you'll be acknowledged in `Authors@R` and given write permission on GitHub.
+> Write permission gives you the power to change {package} directly (which is fine for smaller fixes) but I'd appreciate it if you'd continue to send major stuff through the pull request process.
+>
+> If you accept, can you please prepare a PR that:
+>
+> -   Adds your info to `Authors@R`
+> -   Tweaks `_pkgdown.yml` if you want to link somewhere from the pkgdown site
+> -   Advertises the change in NEWS.md
+> -   Re-builds the documentation to get updated package docs
+>
+> I'll then add you as admin, and approve the PR, then you can squash-merge it.
+> This would be our workflow going forward.
+> (You'll also be able to request reviews from me and other authors as needed).
+>
+> Thanks for all your work on {package}!
+>
+> {your_name}
+
+(After this tidyup is approved, the template will become a usethis function.)
+
+## Open issues
+
+-   How can we communicate with authors? Do we add them to a team? Do we add them as members of the repo? Do we add them to a slack channel?
```

Comment:
Team discussions could potentially work as well if people would prefer that https://docs.github.com/en/organizations/collaborating-with-your-team/creating-a-team-discussion

Though I do think having a slack channel with all the authors across our repos would probably be the best bet.

Reply:
I worry about adding a new communication channel (team discussions) that we're not already using.

Reply:
Yeah, me too. I'll make a note of this concern in the writeup.